zk/conn.go

```diff
@@ -267,7 +284,7 @@ func (c *Conn) loop() {
 
 		// Yeesh
 		if err != io.EOF && err != ErrSessionExpired && !strings.Contains(err.Error(), "use of closed network connection") {
-			log.Println(err)
+			c.logger.Printf("%s\n", err)
```

Comment:
The formatting is there to retain the Newline character that `Println()` adds automatically. I was trying to retain the same behavior.

If you aren't worried about that final character I'll remove the formatting.